ext/bcmath/libbcmath/src/compare.c

```diff
@@ -45,110 +41,103 @@
+				return (1);
+			} else {
+				return (-1);
+			}
+		} else {
+			/* Magnitude of n1 < n2. */
+			if (!use_sign || n1->n_sign == PLUS) {
+				return (-1);
+			} else {
+				return (1);
+			}
+		}
 	}
-    }
-
-  /* If we get here, they have the same number of integer digits.
-     check the integer part and the equal length part of the fraction. */
-  count = n1->n_len + MIN (n1->n_scale, n2->n_scale);
-  n1ptr = n1->n_value;
-  n2ptr = n2->n_value;
-
-  while ((count > 0) && (*n1ptr == *n2ptr))
-    {
-      n1ptr++;
-      n2ptr++;
-      count--;
-    }
-  if (ignore_last && count == 1 && n1->n_scale == n2->n_scale)
-    return (0);
-  if (count != 0)
-    {
-      if (*n1ptr > *n2ptr)
-	{
-	  /* Magnitude of n1 > n2. */
-	  if (!use_sign || n1->n_sign == PLUS)
-	    return (1);
-	  else
-	    return (-1);
+
+	/* If we get here, they have the same number of integer digits.
+	   check the integer part and the equal length part of the fraction. */
+	size_t count = n1->n_len + MIN (n1->n_scale, n2->n_scale);
+	n1ptr = n1->n_value;
+	n2ptr = n2->n_value;
+
+	while ((count > 0) && (*n1ptr == *n2ptr)) {
+		n1ptr++;
+		n2ptr++;
+		count--;
 	}
-      else
-	{
-	  /* Magnitude of n1 < n2. */
-	  if (!use_sign || n1->n_sign == PLUS)
-	    return (-1);
-	  else
-	    return (1);
+
+	if (ignore_last && count == 1 && n1->n_scale == n2->n_scale) {
+		return (0);
 	}
-    }
-
-  /* They are equal up to the last part of the equal part of the fraction. */
-  if (n1->n_scale != n2->n_scale)
-    {
-      if (n1->n_scale > n2->n_scale)
-	{
-	  for (count = n1->n_scale-n2->n_scale; count>0; count--)
-	    if (*n1ptr++ != 0)
-	      {
-		/* Magnitude of n1 > n2. */
-		if (!use_sign || n1->n_sign == PLUS)
-		  return (1);
-		else
-		  return (-1);
-	      }
+	if (count != 0) {
+		if (*n1ptr > *n2ptr) {
+			/* Magnitude of n1 > n2. */
+			if (!use_sign || n1->n_sign == PLUS) {
+				return (1);
+			} else {
+				return (-1);
+			}
+		} else {
+			/* Magnitude of n1 < n2. */
+			if (!use_sign || n1->n_sign == PLUS) {
+				return (-1);
+			} else {
+				return (1);
+			}
+		}
 	}
-      else
-	{
-	  for (count = n2->n_scale-n1->n_scale; count>0; count--)
-	    if (*n2ptr++ != 0)
-	      {
-		/* Magnitude of n1 < n2. */
-		if (!use_sign || n1->n_sign == PLUS)
-		  return (-1);
-		else
-		  return (1);
-	      }
+
+	/* They are equal up to the last part of the equal part of the fraction. */
+	if (n1->n_scale != n2->n_scale) {
+		if (n1->n_scale > n2->n_scale) {
+			for (count = n1->n_scale-n2->n_scale; count>0; count--) {
+				if (*n1ptr++ != 0) {
+					/* Magnitude of n1 > n2. */
+					if (!use_sign || n1->n_sign == PLUS) {
+						return (1);
+					} else {
+						return (-1);
+					}
+				}
+			}
+		}
+	} else {
+		for (count = n2->n_scale-n1->n_scale; count>0; count--) {
+			if (*n2ptr++ != 0) {
+				/* Magnitude of n1 < n2. */
+				if (!use_sign || n1->n_sign == PLUS) {
+					return (-1);
+				} else {
+					return (1);
+				}
+			}
+		}
```

Comment:
Hey @Girgias! I analyzed your PR and found the cause of the failing tests. Problem is with if/else conditions here. You changed scope of top level `if` in those lines. Top level `if` doesn't have `else` statement.

If you need help with this PR I offer it, because I would like to have this extension tidied up. 🙌🏾

Reply:
Thank you that was very helpful :)

Was busy with trying to get stuff done for feature freeze, so if the rebase, fix + force push has everything working on CI I'll merge this. :)